src/dune_engine/action_builder.mli

```diff
@@ -103,14 +103,26 @@ val paths_existing : Path.t list -> unit t
     action produced by the action builder. *)
 val env_var : string -> unit t
 
+type alias_status =
+  | Exists
+  | Missing
+
 val alias : Alias.t -> unit t
 
-val dep_on_alias_if_exists : Alias.t -> bool t
+val dep_on_alias_if_exists : Alias.t -> alias_status t
+
+module Alias_rec (_ : sig
+  (* This API isn't fully baked yet. We might move it to the rules *)
 
-(** Depend on an alias recursively. Return [true] if the alias is defined in at
-    least one directory, and [false] otherwise. *)
-val dep_on_alias_rec :
-  Alias.Name.t -> Context_name.t -> Source_tree.Dir.t -> bool t
+  (** [exists dir ~f] traverses [dir] and evaluates [f] for every directory.
+      Returns [true] if [f] returned true at least once. *)
```

Comment:
The comment is stale. Also, I feel like with the new `alias_status` type, this function can be called `traverse` or something like that.